ape_etherscan/verify.py

```diff
@@ -0,0 +1,171 @@
+        if spdx_key not in first_line:
+            return license_type
+
+        license_id = first_line.split(spdx_key)[-1].strip().lower()
+        license_type = license_keyword_map.get(license_id, 1)
+        if license_type:
+            return license_type
+
+        logger.warning(f"Unsupported license type '{license_id}'.")
+        return license_type
+
+    def attempt_verification(self):
+        compiler = self.compiler_manager.registered_compilers[self._ext]
+        manifest = self.project_manager.extract_manifest()
+        compiler_used = [
+            c for c in manifest.compilers if self._contract_type.name in c.contractTypes
+        ][0]
+        optimizer = compiler_used.settings.get("optimizer", {})
+        optimized = optimizer.get("enabled", False)
+        runs = optimizer.get("runs", 200)
+        source_name = self._contract_type.source_id
+        sources = {source_name: {"content": manifest.sources[source_name].content}}
+        all_settings = compiler.get_compiler_settings(
+            [self._source_path], base_path=self._base_path
+        )
+        settings = all_settings[Version(compiler_used.version)]
+
+        # TODO: Handle libraries, and metadata.
+        source_code = {
+            "language": compiler.name.capitalize(),
+            "sources": sources,
+            "settings": settings,
+        }
+
+        evm_version = compiler_used.settings.get("evmVersion")
+        guid = self.contract_client.verify_source_code(
+            source_code,
+            compiler_used.version,
+            contract_name=f"{self._contract_type.source_id}:{self._contract_type.name}",
+            optimization_used=optimized,
+            optimization_runs=runs,
+            constructor_arguments=self.constructor_arguments,
+            evm_version=evm_version,
+            license_type=self.license_code,
+        )
+        self._wait_for_verification(guid)
+
+    def _wait_for_verification(self, guid: str):
+        iterations = 0
+        timeout = 100
+        while iterations < 25:
+            verification_update = self.contract_client.check_verify_status(guid)
+            fail_key = "Fail - "
+            pass_key = "Pass - "
+            if verification_update.startswith(fail_key):
+                err_msg = verification_update.split(fail_key)[-1].strip()
+                raise ContractVerificationError(err_msg)
+            elif verification_update == "Already Verified" or verification_update.startswith(
+                pass_key
+            ):
+                uri = self.provider.network.explorer.get_address_url(self.address)
+                logger.success(f"Contract verification successful!\n{uri}#code")
+                break
+
+            status_message = f"Contract verification status: {verification_update}"
+            logger.info(status_message)
+
+            time.sleep(2.5)
+            iterations += 1
+            if iterations == timeout:
```

Comment:
if I'm reading this correctly, this will never be true (iterations cannot exceed 25, while timeout is 100)